components/com_content/models/article.php

```diff
@@ -164,7 +177,7 @@ public function getItem($pk = null)
 				}
 
 				// Check for published state if filter set.
-				if (((is_numeric($published)) || (is_numeric($archived))) && (($data->state != $published) && ($data->state != $archived)))
+				if ( !($haveToken == $currentToken) && ((is_numeric($published)) || (is_numeric($archived))) && (($data->state != $published) && ($data->state != $archived)))
```

Comment:
This is not completely correct, because the article would still load if it is archived. We should only check the token if the article is unpublished.